app/Actions/Photo/Strategies/AddStandaloneStrategy.php

```diff
@@ -147,36 +171,32 @@ protected function normalizeOrientation(): void
 			// Reset source file info to the new temporary and ensure that
 			// it will be deleted later
 			$this->parameters->sourceFileInfo = SourceFileInfo::createByTempFile(
-				$info->getOriginalName(),
-				$info->getOriginalExtension(),
-				$tmpFile
-			);
+					$info->getOriginalName(),
+					$info->getOriginalExtension(),
+					$tmpFile
+				);
 			$this->parameters->importMode->setDeleteImported(true);
 		}
 
-		/** @var ImageHandlerInterface $imageHandler */
-		$imageHandler = resolve(ImageHandlerInterface::class);
-
 		$absolutePath = $this->parameters->sourceFileInfo->getFile()->getAbsolutePath();
 		// If we are importing via symlink, we don't actually overwrite
-		// the source but we still need to fix the dimensions.
-		$newDim = $imageHandler->autoRotate(
+		// the source, but we still need to fix the dimensions.
+		$this->imageHandler->autoRotate(
 			$absolutePath,
 			$orientation,
 			$this->parameters->importMode->shallImportViaSymlink()
 		);
 
-		if ($newDim !== [false, false]) {
-			// If the image has actually been rotated, the size
-			// and the checksum may have changed.
-			/* @var  Extractor $metadataExtractor */
-			$metadataExtractor = resolve(Extractor::class);
-			$this->photo->filesize = $metadataExtractor->filesize($absolutePath);
-			$this->photo->checksum = $metadataExtractor->checksum($absolutePath);
-			$this->photo->save();
-		}
+		// If the image has actually been rotated, the size
+		// and the checksum may have changed.
+		$this->photo->filesize = $this->metadataExtractor->filesize($absolutePath);
+		$this->photo->checksum = $this->metadataExtractor->checksum($absolutePath);
+		$this->photo->save();
```

Comment:
For all of the mentioned reasons.

1. If `rotate` _fails_, then it throws a proper exception. `rotate` doesn't return `[false, false]`.
2. If `rotate` is a _no op_, i.e. the photo does not require rotation because it is already correctly oriented, then `rotate` returns an array with the original dimensions, but not `[false, false]`.
3. `->save` calculates the changed set of attributes and does nothing, if there was no change.

In other words, the condition is never taken and in case of a no-op, `save` itself prevents an unnecessary DB operation.